go-controller/pkg/ovn/ovn.go

```diff
@@ -304,6 +304,10 @@ func (oc *Controller) Run(wg *sync.WaitGroup, nodeName string) error {
 	klog.Infof("Starting all the Watchers...")
 	start := time.Now()
 
+	// Sync external gateway routes. External gateway may be set in namespaces
+	// or via pods. So execute an individual sync method at startup
+	oc.cleanExGwECMPRoutes()
```

Comment:
I dont think so. The delete path is "best effort. So if we delete a bunch of routes for pods that are about to be deleted anyway, there should be no conflict. In other words, pod deletion should try as best as it can to remove the ECMP routes, but if it cant, it shouldn't fail and should clean up the rest of the pod.